### DISCUSSION TITLE
Steps to migrate from Brew under Rosetta2 to native Brew 3.0?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#707] [Getting started] [answered]
What is the process to migrate from having installed Brew (and applications via Brew) under Rosetta2 on Apple Silicon to native ARM Brew 3.0?

**Comment by SMillerDev** [comment, marked as answer]
You install them again using the native install. There is no easy migration path because not all software works on ARM yet.

https://github.com/Homebrew/homebrew-bundle can help you ensure you have all the packages in both installs though.

---
see also https://github.com/Homebrew/discussions/discussions/417 https://github.com/Homebrew/discussions/discussions/437 https://github.com/Homebrew/discussions/discussions/545 https://github.com/Homebrew/discussions/discussions/357

**Reply by SMillerDev** [reply]
A native install of brew is a different thing than a rosetta 2 install of brew. They are managed completely separately because they're essentially running on a different OS.

If you want to get rid of Rosetta packages, you need to uninstall them

**Comment by adamwinn** [comment]
brew.sh says `This script installs Homebrew to its preferred prefix (/usr/local for macOS Intel, /opt/homebrew for Apple Silicon)`

I'm on an m1 mac.  I paste in the install command and it wants to install into `/usr/local`

```
bash-3.2$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
```

Why isn't it wanting to install into `/opt/homebrew`?

**Reply by carlocab** [reply]
Are you running your terminal under Rosetta?

**Reply by adamwinn** [reply]
I'm running iTerm and Activity Monitor says it has the Apple architecture

**Reply by adamwinn** [reply]
I'm guessing `Rosetta 2: true` is why. How can I change that?

```
brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 10 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 8bf0ee917c5b5f2de1aecec375f3ae37c70db5e4
Core tap last commit: 24 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 12.0 build 1200
Git: 2.30.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 1.8.0_51
macOS: 11.2-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: true
```

**Reply by andrewcrook** [reply]
"get info" from finder on item icon. There is a check box "open in Rosetta".

**Reply by andrewcrook** [reply]
You can check by using the "arch" command.

**Comment by atze007** [comment]
I got the same Problem. I am nor using the Rosetta Terminal, but it ist still the `/usr/local` folder. What can I do to change that an use the native ARM brew?

brew config
HOMEBREW_VERSION: 3.0.0-11-g28fb4ff
ORIGIN: https://github.com/Homebrew/brew
HEAD: 28fb4ff6437efd24a97123ea69caee47ad400a7d
Last commit: 20 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: b024c23fcfcc26440a08dd0d325b289577d5617e
Core tap last commit: 20 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.78KUNmz6DF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.23.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: false

**Reply by resistorsoftware** [reply]
I was confused by this too. Realized I had to open an iTerm not set to Rosetta, ensured Activity monitor showed apple and not intel, then ran  the install. Instead of /usr/local I was presented with /opt/homebrew. So it seems to work and install properly.

**Reply by atze007** [reply]
It seems to be already installed in the /opt/homebrew, at least I can find a folder with the respective files...
But when I try a brew command, it is always referring to the /usr/local ...

**Reply by andrewcrook** [reply]
Run "arch" what does it say?

**Reply by atze007** [reply]
arm64

**Reply by carlocab** [reply]
You must've been running under Rosetta when you installed `brew`. Run the installer again; make sure you're **not** running in Rosetta when you do it.

This will install a separate native installation in `/opt/homebrew`.

**Reply by atze007** [reply]
if I do that i get:

...
From https://github.com/Homebrew/brew
   28fb4ff64..9e1f35525  master     -> origin/master
HEAD is now at 9e1f35525 Merge pull request #10544 from Homebrew/sorbet-files-update
Warning: git 2.30.0 is already installed and up-to-date.
To reinstall 2.30.0, run:
  brew reinstall git
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63289 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63302 Killed: 9               "$GIT_EXECUTABLE" "$@"
Error: Fetching /opt/homebrew failed!
Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core failed!
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63316 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63324 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63325 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63343 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 63344 Killed: 9               "$GIT_EXECUTABLE" "$@"
Error: update-report should not be called directly!
Failed during: /opt/homebrew/bin/brew update --force --quiet

**Reply by atze007** [reply]
And if i try brew reinstall git:

Error: Cannot install in Homebrew on ARM processor in Intel default prefix (/usr/local)!
Please create a new installation in /opt/homebrew using one of the
"Alternative Installs" from:
  https://docs.brew.sh/Installation
You can migrate your previously installed formula list with:
  brew bundle dump

**Reply by carlocab** [reply]
What do the following commands say:
```
which -a brew
/usr/local/bin/brew config
/opt/homebrew/bin/brew config
which -a git
```

**Reply by atze007** [reply]
which -a brew
/usr/local/bin/brew
/opt/homebrew/bin/brew

/usr/local/bin/brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a094e3df380a4788574d6d97ecf46952da9348f0
Core tap last commit: 11 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.78KUNmz6DF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.23.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: false

/opt/homebrew/bin/brew config
HOMEBREW_VERSION: >=2.5.0 (shallow or no git repository)
ORIGIN: (none)
HEAD: (none)
Last commit: never
Core tap ORIGIN: (none)
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: (none)
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.78KUNmz6DF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: N/A
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: false

which -a git
/usr/local/bin/git
/usr/bin/git
/opt/homebrew/bin/git

**Reply by carlocab** [reply]
You need to put `/opt/homebrew/bin` earlier in your `PATH`, so that calls to `brew` call your `/opt/homebrew` installation and not the one in `/usr/local`.

Once you've done that, do
```
/opt/homebrew/bin/brew uninstall git
/opt/homebrew/bin/brew update
```
You can reinstall `git` after doing the `brew update`. You don't need to include the `/opt/homebrew/bin` part if you've fixed your `PATH` correctly.

**Reply by atze007** [reply]
thanks a lot! Now it is calling the `opt/homebrew` but it seems that the 3.0.0 Version is not installed:

brew config
HOMEBREW_VERSION: >=2.5.0 (shallow or no git repository)
ORIGIN: (none)
HEAD: (none)
Last commit: never
Core tap ORIGIN: (none)
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: (none)
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.78KUNmz6DF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: N/A
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: false

I can't update with `brew update`

brew update
Warning: git 2.30.0 is already installed and up-to-date.
To reinstall 2.30.0, run:
  brew reinstall git
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91674 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91685 Killed: 9               "$GIT_EXECUTABLE" "$@"
Error: Fetching /opt/homebrew failed!
Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core failed!
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91699 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91707 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91708 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91726 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18: 91727 Killed: 9               "$GIT_EXECUTABLE" "$@"
==> Homebrew has enabled anonymous aggregate formula and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics have been recorded yet (nor will be during this `brew` run).

Error: Failure while executing; `git config --replace-all homebrew.analyticsmessage true` was terminated by uncaught signal KILL.

**Reply by atze007** [reply]
Now I used the `brew uninstall git` command again and I get the following by using `brew config`

brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e12f98fd2fc54bcde716bf76fe2bdf280225eb62
Core tap last commit: 28 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.78KUNmz6DF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: false

Seems fine, right?

**Reply by atze007** [reply]
But when I `brew install git` I get the wired `brew Config` from the post before....

**Reply by carlocab** [reply]
Hmmm, yes, that is quite weird. Can you do
```
/opt/homebrew/bin/brew uninstall git gettext pcre2
/opt/homebrew/bin/brew install git
/opt/homebrew/bin/brew config
```
Then tell me the output of `brew config`?

**Reply by atze007** [reply]
brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f4b1eacad3e6c37423e85fdd7b5d0e5a0f27dc75
Core tap last commit: 7 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.78KUNmz6DF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.30.0 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: false

**Comment by adamwinn** [comment]
My iTerm is not set to Rosetta and Activity Monitor says Apple and it still wants to install into /usr/local

It doesn't even work using the macOS Terminal

**Reply by resistorsoftware** [reply]
Are you sure you are entering a command in an iterm that is apple? You could have 2 iterms active?

**Reply by carlocab** [reply]
Try putting `arch -arm64e` right before the install command, so it should look something like
```
arch -arm64e /bin/bash $(curl ...)
```

**Reply by adamwinn** [reply]
That worked, thanks!

**Reply by andrewcrook** [reply]
If you run just "arch" and it says it's not ARM64 then your terminal program or shell is running via Rosetta 2
